src/opensignals/data/provider.py

```diff
@@ -0,0 +1,257 @@
+        ).dt.strftime('%Y-%m-%d')
+        tickers_outdated.drop(columns=['date_max'], inplace=True)
+
+        return pd.concat(
+            [ticker_not_found, tickers_outdated]
+        )
+
+    @staticmethod
+    def get_live_data(ticker_data,  last_friday):
+        date_string = last_friday.strftime('%Y-%m-%d')
+        live_data = ticker_data[ticker_data.date == date_string].copy()
+
+        # get data from the day before, for markets that were closed
+        last_thursday = last_friday - relativedelta(days=1)
+        thursday_date_string = last_thursday.strftime('%Y-%m-%d')
+        thursday_data = ticker_data[ticker_data.date == thursday_date_string]
+
+        # Only select tickers than aren't already present in live_data
+        thursday_data = thursday_data[
+            ~thursday_data.bloomberg_ticker.isin(live_data.bloomberg_ticker.values)
+        ].copy()
+
+        live_data = pd.concat([live_data, thursday_data])
+        live_data = live_data.set_index('date')
+        return live_data
+
+    @staticmethod
+    def get_train_test_data(ticker_data, feature_names, targets):
+        """merge our feature data with Numerai targets"""
+        ml_data = pd.merge(
+            ticker_data, targets,
+            on=['date', 'bloomberg_ticker'],
+            how='left'
+        )
+
+        logger.info(f'Found {ml_data.target.isna().sum()}'
+                    'rows without target, filling with 0.5')
+        ml_data['target'] = ml_data['target'].fillna(0.5)
+
+        # convert date to datetime and index on it
+        ml_data = ml_data.set_index('date')
+
+        # for training and testing we want clean, complete data only
+        ml_data = ml_data.dropna(subset=feature_names)
+        # ensure we have only fridays
+        ml_data = ml_data[ml_data.index.weekday == 4]
+        # drop eras with under 50 observations per era
+        ml_data = ml_data[ml_data.index.value_counts() > 50]
+
+        # train test split
+        train_data = ml_data[ml_data['data_type'] == 'train']
+        test_data = ml_data[ml_data['data_type'] == 'validation']
+        return train_data, test_data
+
+    def get_data(self,
+                 db_dir,
+                 features_generators=None,
+                 last_friday=datetime.today() - relativedelta(weekday=FR(-1)),
+                 target='target_20d',
+                 feature_prefix=None):
+        """generate data set"""
+
+        if features_generators is None:
+            features_generators = []
+
+        ticker_data = self.get_ticker_data(db_dir)
+
+        ticker_universe = pd.read_csv(SIGNALS_UNIVERSE)
+        ticker_data = ticker_data[ticker_data.bloomberg_ticker.isin(
+            ticker_universe['bloomberg_ticker'])]
+
+        targets = pd.read_csv(SIGNALS_TARGETS)
+        targets['date'] = pd.to_datetime(
+            targets['friday_date'],
+            format='%Y%m%d'
+        )
+        targets['target'] = targets[target]
+
+        feature_names = []
```

Comment:
with the features prefix we might no longer need to explicitly collect the feature names, but I guess that is outside the scope of this PR

Reply:
Agreed